### DISCUSSION TITLE
Plugin to get to bottom or back to top of a page

### DISCUSSION THREAD

**Discussion by dinojr** [xournalpp/xournalpp#4701] [Q&A]
I didn't find a key/menu entry to get to the bottom and to the top of the page. Could it be achieved with a plugin ?

**Comment by rolandlo** [comment]
I think so, but haven't tried.
There is `app.scrollToPos()` which can be used both in absolute and relative mode.

https://github.com/xournalpp/xournalpp/blob/b96038561c7a369d390c644ccefad1389c9de1c3/src/core/plugin/luapi_application.h#L1648-L1658

So 
```lua
app.scrollToPos(0,0,false)
``` 
should scroll to the page top and `
```lua
app.scrollToPos(0,0,false)
app.scrollToPos(0,height)
```
where `height` is the page height (in pt) should scroll to the bottom.
Note that you can retrieve the page height using

```lua
local docStructure = app.getDocumentStructure()`
local currentPage = docStructure["currentPage"]
local height = docStructure["pages"][currentPage]["pageHeight"]
```
